tasks/trace_agent.py

```diff
@@ -67,7 +67,10 @@ def build(
     cmd = f"go build -mod={go_mod} {race_opt} {build_type} -tags \"{go_build_tags}\" "
     cmd += f"-o {agent_bin} -gcflags=\"{gcflags}\" -ldflags=\"{ldflags}\" {REPO_PATH}/cmd/trace-agent"
 
-    ctx.run(f"go generate -mod={go_mod} {REPO_PATH}/pkg/trace/info", env=env)
+    # go generate only works if you are in the module the target file is in, so we
+    # need to move into the pkg/trace module.
+    with ctx.cd("./pkg/trace"):
```

Comment:
Is this assuming that you are running this command in a specific directory? Otherwise I assume using a relative path here may cause problems.

Reply:
> Yeah I believe the ctx here refers to the root directory of the agent as a whole

Yes, the invoke tasks have to be run from the root directory of the `datadog-agent` repository. Therefore the initial working directory of the invoke context is that root directory, and `pkg/trace` should always exist relative to that directory.